### PR TITLE
feat(api): add exclude_defaults param for post with refs

### PR DESCRIPTION
**Pull request by irynadosiak-c-change-lab**

### CODE REVIEW

File: src/openepd/api/average_dataset/generic_estimate_sync_api.py [hunk 1]
```diff
@@ -57,26 +57,33 @@ def get_by_openxpd_uuid(
...
 
     @overload
-    def post_with_refs(self, ge: GenericEstimateWithDeps, with_response: Literal[False] = False) -> GenericEstimate: ...
+    def post_with_refs(
+        self, ge: GenericEstimateWithDeps, with_response: Literal[False] = False, exclude_defaults: bool = True
+    ) -> GenericEstimate: ...
 
     def post_with_refs(
-        self, ge: GenericEstimateWithDeps, with_response: bool = False
+        self, ge: GenericEstimateWithDeps, with_response: bool = False, exclude_defaults: bool = True
```
Comment (corvis) [inline src/openepd/api/average_dataset/generic_estimate_sync_api.py:69]: Why we do this only for generic estimates, what about epds?

Comment (irynadosiak-c-change-lab) [reply src/openepd/api/average_dataset/generic_estimate_sync_api.py:69]: My mistake, I'll update

Comment (irynadosiak-c-change-lab) [reply src/openepd/api/average_dataset/generic_estimate_sync_api.py:69]: updated

File: src/openepd/api/utils.py [hunk 2]
```diff
@@ -26,3 +26,18 @@ def encode_path_param(value: str) -> str:
     :return: encoded value
     """
     return quote(value, safe="")
+
+
+def remove_none_id_fields(d: dict) -> dict:
+    """Remove any key 'id' with a None value from the dictionary, including nested dicts."""
```
Comment (alexlitvinenko) [inline src/openepd/api/utils.py:32]: Describe args and return type, say something about absence of side effects.

Comment (irynadosiak-c-change-lab) [reply src/openepd/api/utils.py]: updated

**Review by alexlitvinenko — APPROVED** [review]

**Pull request merged**